ops/framework.py

```diff
@@ -545,14 +549,19 @@ class Framework(Object):
     """Main interface from the Charm to the Operator Framework internals."""
 
     on = FrameworkEvents()  # type: ignore
+    """Used for :meth:`observe`-ing framework-specific events."""
 
     # Override properties from Object so that we can set them in __init__.
     model: 'Model' = None  # type: ignore
+    """The :class:`Model` instance for this charm."""
+
     meta: 'CharmMeta' = None  # type: ignore
+    """The charm's metadata."""
+
     charm_dir: 'pathlib.Path' = None  # type: ignore
+    """The top-level directory of the charm."""
```

Comment:
how about "charm project root directory"? We call it 'charm_root' in scenario.

Reply:
Good call -- fixed